app/controllers/DisclosureDetailsController.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.actions._
+import javax.inject.Inject
+import pages.disclosure.DisclosureIdentifyArrangementPage
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import renderer.Renderer
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+
+import scala.concurrent.ExecutionContext
+
+class DisclosureDetailsController @Inject()(
+    override val messagesApi: MessagesApi,
+    identify: IdentifierAction,
+    getData: DataRetrievalAction,
+    val controllerComponents: MessagesControllerComponents,
+    renderer: Renderer
+)(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
+
+  def onPageLoad: Action[AnyContent] = (identify andThen getData).async {
+    implicit request =>
+
+      val arrangementMessage = if (request.userAnswers.isDefined){
```

Comment:
Or:

`  val arrangementMessage: String = ua.fold("") { value =>
    value.get(DisclosureIdentifyArrangementPage).map(msg"disclosureDetails.forArrangement".withArgs(_).resolve).getOrElse("")
  }
  // disclosureDetails.forArrangement = for arrangement {0}
`